haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -189,8 +190,11 @@ def warm_up(self):
         """
         Initializes the component.
         """
-        if self.pipeline is None:
-            self.pipeline = pipeline(**self.huggingface_pipeline_kwargs)
```

Comment:
This is a bit of an exception. Technically the warmup state can be determined by the value of `self.pipeline` alone but this is not always true for all the HF-based components, and I'm not sure why. For consistency I went with the `self._warmed_up` strategy for all the HF family, but I can revert this if it's better.